### ISSUE TITLE
Change font on popups

### ISSUE THREAD

**Issue by krstp** [jisaacks/GitGutter#577]
I would like to have an option to change the font on the inline blame popups as well as on the window popups. I tried to toy with the `css` popup stylesheet file, but no success :(

In particular what is annoying, for some reason it picks up by default a Times font, which definitely is not my prime choice:
![inline](https://github.com/jisaacks/GitGutter/assets/4711498/557623d5-d49b-44d6-832b-77df0c035265)

I would like to be able to change the font to my liking.

**Comment by deathaxe** [comment]
"Blame" uses ST's annotation API. GitGutter doesn't assign explicit font-family settings to it. The font is choosen by ST or any CSS embedded in the active color scheme.

By default ST should pick the font from preferences. I already saw Times New Roman being applied, too, but never found out the root cause.

Font name is a global setting, which would effect any plugin using annotations api. Hence it doesn't make sense to introduce a font-family setting to adjust it for GitGutter only.

Diff popup also uses a view's font to display diffs. Font is set only for toolbar glyphs. Diff popup can be modified via `Preferences: GitGutter Popup Stylesheet`.

I am able to adjust "diff" fonts via:

```css
/* GitGutter Popup Stylesheet - User */

.git-gutter div.highlight {
    font-family: Arial;
}
```

It not working on your end, may be caused by the same reason for Times New Roman being selected for annotations.

**Comment by krstp** [comment]
So here is an interesting thing...
I can change global font in the settings via user preferences:
```
{
    "font_face": "MesloLGS NF",
//  "font_face": "agave NF",
}
```
which will pick up the desired font and change it globally, but inline/popups will NOT accept it as if it cannot find the font (which seems to be consistent with other posts on the web having similar issue). Note, all the fonts are instanced in custom font Mac path, so should be visible.

When I comment the the line out `//` then of course Sublime picks up the default font (whatever it is), and here the inline/poups appear to be catching the same global font without "Times-ing" it, but it just means I cannot use the custom font 🤔 

So I am kind of clueless why would custom font work globally except popups and inline 🤔 As if ST is not picking up the new custom-default font.

**Comment by krstp** [comment]
Yeah, and I did try diff options for GitGutter and I do not see any relevant changes :(

**Comment by deathaxe** [comment]
That's however nothing related to GitGutter.

Maybe another package or color scheme messes around with `font-family` property in global manner, preventing ST from applying its defaults (global `font_face`). Proposed font-family changes of popup dialog not being picked up is a rather reliable indicator for that happening. So there's nothing GitGutter could do to work around that.


ST has known limitations with regards to correctly cascade multiple stylesheets or picking up changes from color_scheme's CSS settings.

Related core issues are:
- https://github.com/sublimehq/sublime_text/issues/2935
- https://github.com/sublimehq/sublime_text/issues/3650

_Notes:_

_All popups/annotations pick up correct fonts on my machine._

_Diff Popups are rendered by `mdpopups` library._

**Comment by krstp** [comment]
Without in-depth testing, this is the assumption and wild guess I make, so thanks for verifying. When I have a minute I might look more into it, for now I guess the default font is fine for me. Thanks for following up 🙏
